src/wp-admin/includes/image.php

```diff
@@ -1143,3 +1268,58 @@ function _copy_image_file( $attachment_id ) {
 
 	return $dst_file;
 }
+
+/**
+ * Returns an array with the list of valid mime types that a specific mime type should be converted into.
+ * For example an `image/jpeg` should be converted into an `image/jpeg` and `image/webp`. The first type
+ * is considered the primary output type for this image.
+ *
+ * @since 6.0.0
+ *
+ * @param $attachment_id int The attachment ID.
+ * @return array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function wp_upload_image_mime_transforms( $attachment_id ) {
+	$image_mime_transforms = array(
+		'image/jpeg'      => array( 'image/jpeg', 'image/webp' ),
+		'image/webp'      => array( 'image/webp', 'image/jpeg' ),
+		'application/pdf' => array( 'image/jpeg' ),
+	);
+
+	/**
+	 * Filter to the output mime types for a given input mime type.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $image_mime_transforms A map with the valid mime transforms.
+	 * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
+	 */
+	return (array) apply_filters( 'wp_upload_image_mime_transforms', $image_mime_transforms, $attachment_id );
+}
+
+/**
+ * Extract the primary and additional mime output types for an image from the $image_mime_transforms.
+ *
+ * @since 6.0.0
+ *
+ * @param $image_mime_transforms array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ * @param $attachment_id int The attachment ID.
+ * @param $mime_type string The mime type of the image.
+ */
+function _wp_get_primary_and_additional_mime_types( $image_mime_transforms, $attachment_id, $mime_type ) {
+	$mime_type         = get_post_mime_type( $attachment_id );
```

Comment:
This seems off I think? Why pass `$mime_type` if it is immediately overwritten? Also, I think it would be good to use something like `$original_mime_type` as name.

That mime type parameter is required, so I think we should also get rid of passing the `$attachment_id` here. Last but not least, we shouldn't rely on `get_post_mime_type` for this in the first place as that is not necessarily the MIME type of the image to process, e.g. in the PDF scenario. See also https://github.com/WordPress/wordpress-develop/pull/2393#discussion_r828483337

Reply:
👍🏼  excellent points, will update

Reply:
Updated, and as expected this fixed the issue with pdf mapping. The system now generates webp versions of the images it generates jpegs for **except the full sized** image since core has special handling for that. We may need to add a loop there for additional mime types as well.